src/tiled/templatesdock.cpp

```diff
@@ -300,6 +371,54 @@ void TemplatesDock::retranslateUi()
     mChooseDirectory->setText(tr("Choose Templates Directory"));
 }
 
+void TemplatesDock::fixTileset()
+{
+    Q_ASSERT(mObject);
+
+    Cell cell = mObject->cell();
+
+    if (cell.isEmpty())
+        return;
+
+    auto tileset = cell.tileset()->sharedPointer();
+
+    if (tileset->imageStatus() == LoadingError) {
+        // This code opens a new document even if there is a tileset document
+        auto tilesetDocument = DocumentManager::instance()->findTilesetDocument(tileset);
+        if (!tilesetDocument) {
+            tilesetDocument = new TilesetDocument(tileset, tileset->fileName());
+            DocumentManager::instance()->addDocument(tilesetDocument);
+        } else {
+            DocumentManager::instance()->openTileset(tileset);
+        }
+        connect(tilesetDocument, &TilesetDocument::tilesetChanged,
+                this, &TemplatesDock::applyChanges);
```

Comment:
This connection is a little strange, because the template didn't change when the tileset was changed. Would it work to only connect this to `checkTileset`?

Also, clicking the button multiple times would make multiple connections. I think we can solve that by passing `Qt::UniqueConnection`.